gateway/src/services/config-manager-cert-passphrase.ts

```diff
@@ -0,0 +1,46 @@
+import fs from 'fs';
+import yaml from 'js-yaml';
+import { logger } from './logger';
+
+export namespace ConfigManagerCertPassphrase {
+  export const passphraseFliePath: string = './conf/gateway-passphrase.yml';
+
+  export interface PassphraseConfig {
+    CERT_PASSPHRASE: string;
+  }
+
+  export function readPassphrase(): string {
+    if (fs.existsSync(passphraseFliePath)) {
+      const mode = fs.lstatSync(passphraseFliePath).mode;
+      // check and make sure the passphrase file is a regular file, and is only accessible by the user.
+      if (
+        mode ===
+        (fs.constants.S_IFREG | fs.constants.S_IWUSR | fs.constants.S_IRUSR)
+      ) {
+        const x = yaml.load(
+          fs.readFileSync(passphraseFliePath, 'utf8')
+        ) as PassphraseConfig;
+        if ('CERT_PASSPHRASE' in x) {
+          return x.CERT_PASSPHRASE;
+        } else {
+          logger.error(
+            passphraseFliePath + ' does not have CERT_PASSPHRASE set.'
+          );
+          process.exit(1);
```

Comment:
Since this is the very point of this module, we should add a way to make this testable. We can't test the process exit itself of course, but we can test whether it's place in the code has been triggered.

We can do this by adding a layer of indirection. e.g.

```
export const bindings = {
   _exit: process.exit,
};

        ...
        logger.error(
          passphraseFliePath +
            ' file does not strictly have mode set to user READ_WRITE only.'
        );
        bindings._exit(1);
        ...
```

So when it comes to unit testing, we can then use the `patch()` function to test the exit functionality.

```
  ...
  patch(ConfigManagerCertPassphrase.bindings, "_exit", () => { /* record something here */ });
  ...
```